controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -320,6 +320,45 @@ public void deleteStream(final String scopeName, final String streamName, final
         .thenAccept(x -> LoggerHelpers.traceLeave(log, "deleteStream", traceId));
     }
 
+    @Override
+    public void deleteReaderGroup(final String scopeName, final String readerGroupName,
+                                  final SecurityContext securityContext, final AsyncResponse asyncResponse) {
+        long traceId = LoggerHelpers.traceEnter(log, "deleteReaderGroup");
+        long requestId =  requestIdGenerator.nextLong();
+
+        try {
+            restAuthHelper.authenticateAuthorize(
+                    getAuthorizationHeader(),
+                    authorizationResource.ofReaderGroupInScope(scopeName, readerGroupName),
+                    READ_UPDATE);
+        } catch (AuthException e) {
+            log.warn(requestId, "delete reader group for {} failed due to authentication failure.",
+                    scopeName + "/" + readerGroupName);
+            asyncResponse.resume(Response.status(Status.fromStatusCode(e.getResponseCode())).build());
+            LoggerHelpers.traceLeave(log, "deleteReaderGroup", traceId);
+            return;
+        }
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(scopeName, this.localController, this.clientConfig);
```

Comment:
Question: creating a `ClientFactoryImpl` and a `ReaderGroupManager` on a per-call basis is expensive (aslo, you don't seem to be closing the resources). I see that in the previous call, deleting a Stream is done by `controllerService.deleteStream()` call. Why don't you do just the same calling to `deleteReaderGroup` in `ControllerService`?

Reply:
`controllerService.deleteReaderGroup()` require readerGroupId along with readerGroupName. This command is taking only readerGroupName as params (not reader group id). As we are already using ReaderGroupManager(to describe reader group) in this class, so i followed up the same.

>  you don't seem to be closing the resources
Closing the resources at Line355
```
thenAccept(response -> {
            asyncResponse.resume(response);
            readerGroupManager.close();
            clientFactory.close();
            LoggerHelpers.traceLeave(log, "deleteReaderGroup", traceId);
        })
```

Reply:
I see, the call requires `readerGroupID` that seems to be stored in the `StateSyncrhonizer` of the `ReaderGroup`. This makes this call quite restrictive in its use, I think. Now I understand why you are using `ClientFactoryImpl` and `ReaderGroupManager` here. Anyway, it may not be ideal but it seems the only approach here, unless we change the  `controllerService.deleteStream()` call or overload it with a version that does not require reader group id.